packages/lit-dev-content/site/docs/frameworks/react.md

```diff
@@ -0,0 +1,208 @@
+import {MyElement} from './my-element.js';
+
+export const MyElementComponent = createComponent({
+  tagName: 'my-element',
+  elementClass: MyElement,
+  react: React,
+  events: {
+    onactivate: 'activate',
+    onchange: 'change',
+  },
+});
+```
+
+After defining the React component, you can use it just as you would any other React component.
+
+```jsx
+<MyElementComponent
+  active={isActive}
+  onactivate={(e) => setIsActive(e.active)}
+  onchange={handleChange}
+/>
+```
+
+{% aside "positive" "no-header" %}
+
+See it in action in the [React playground examples](/playground/#sample=examples/react-basics).
+
+{% endaside %}
+
+#### Options
+
+`createComponent` takes an options object with the following properties:
+
+- `tagName`: The custom element's tag name.
+- `elementClass`: The custom element class.
+- `react`: The imported `React` object. This is used to create the wrapper component with the user supplied `React`. This can also be an import of `preact-compat`.
+- `events`: An object that maps an event handler prop to an event name fired by the custom element.
+
+#### Using slots
+
+Children of component created with `createComponent()` will render to the default slot of the custom element.
+
+```jsx
+<MyElementComponent>
+  <p>This will render in the default slot.</p>
+</MyElementComponent>
+```
+
+To render the child to a specific named slot, the standard `slot` attribute can be added.
+
+```jsx
+<MyElementComponent>
+  <p slot="foo">This will render in the slot named "foo".</p>
+</MyElementComponent>
+```
+
+Since React components are not themselves HTML elements, they usually cannot directly have a `slot` attribute. To render into a named slot, the component will need to be wrapped with a container element that has a `slot` attribute. If a wrapper element interferes with styling, like for grid and flexbox layouts, giving it a `display: contents;` style ([See MDN for details](https://developer.mozilla.org/en-US/docs/Web/CSS/display#box)) will remove the container from rendering, and only render its children.
+
+```jsx
+<MyElementComponent>
+  <div slot="foo" style="display: contents;">
+    <ReactComponent />
+  </div>
+</MyElementComponent>
+```
+
+{% aside "positive" "no-header" %}
+
+Try it out in the [React slots playground example](/playground/#sample=examples/react-slots).
+
+{% endaside %}
+
+#### Events
+
+The `events` option takes an object whose key will be the React prop name and value will be the event name. The function passed to the created React component as a prop that matches the key name will be called when the specified event is fired from the custom element. While the the React prop name can be whatever you want, the recommended convention is to add `on` in front of the event name. This is in line with how React is planning to implement event support for custom element properties. You should also make sure this prop name does not collide with any existing properties on the element.
```

Comment:
This paragraph is a bit long and feels like it combines a few topics. Maybe something like this, or tweak as best you can:

```suggestion
The `events` option takes an object maps React prop names to event names. When a component user passes a callback prop with one of the event prop names, the wrapper will add it as an event handler for the corresponding event.

While the the React prop name can be whatever you want, the recommended convention is to add `on` in front of the event name. This matches how React is planning to implement event support for custom elements. You should also make sure this prop name does not collide with any existing properties on the element.
```